src/main/java/seedu/address/model/property/UniqueBuyerList.java

```diff
@@ -65,4 +66,48 @@ public void setBuyer(Buyer target, Buyer editedBuyer) {
             throw new BuyerNotFoundException();
         }
     }
+
+    /**
+     * Sorts the list by price in ascending order. If two {@code Buyer} have the same price,
+     * the {@code Buyer} which has a lexically smaller name will have a lower priority.
+     */
+    public void sortPrice() {
+        Comparator<Buyer> priceComparator = (b1, b2) -> {
+            if (b1.getMaxPrice().compareTo(b2.getMaxPrice()) == 0) {
+                return b1.getName().compareTo(b2.getName());
+            }
+            return b1.getMaxPrice().compareTo(b2.getMaxPrice());
+        };
+        super.sortListables(priceComparator);
+    }
```

Comment:
The sort options object could be potentially used as follows. We implement PriceComparator, NameComparator and some form of Desc/Asc reversal function statically in the class to be chained.
Suppose the sort options consist of `{ SortType: Price, SortDir: Desc }`
Then the PriceComparator is chained with NameComparator, which is then chained with the reversal function, somewhat like a builder pattern.

Reply:
Perhaps we can have a big SortTypeWorker method to use a switch case statement on the SortType enum and build the SortType component of the final comparator, then a SortDirWorker method to use a switch case statement on the SortDir enum to reverse the comparator if necessary.
Each worker could further delegate the case statements to sub-functions.

Reply:
Hmm its not possible to implement a comparator in the enum class directly as Comparator is generic. What I am thinking of is having the comparators in Property and Buyer classes and call it in the sort function in `UniquePropertyList` and `UniqueBuyerList`.